src/main/java/net/neoforged/neoforge/event/AddReloadListenerEvent.java

```diff
@@ -45,6 +46,20 @@ public void addListener(PreparableReloadListener listener) {
         listeners.add(new WrappedStateAwareListener(listener));
     }
 
+    /**
+     * @param listener the listener to add to the ResourceManager on reload
+     * @param before   a predicate for what listener to add this before
+     */
+    public void addListenerBefore(PreparableReloadListener listener, Predicate<PreparableReloadListener> before) {
```

Comment:
This method should ideally be split into its own PR and account the feedback provided in https://github.com/neoforged/NeoForge/pull/598
@pupnewfster has stated that they don't have updating the PR in their priority list, so if you do end up porting it, you should use a toposort and named listeners.